### PR TITLE
Moved cpe module load position, changed version

### PR DESCRIPTION
**Pull request by RuiApostolo**

### CODE REVIEW

File: apps/GROMACS/ARCHER2_2022.4_gcc11.md [hunk 1]
```diff
@@ -20,10 +20,10 @@ Setup correct modules and build environment
 Setup the GCC programming environment:
 
 ```bash
+module load cpe
```
Comment (juanfrh) [inline apps/GROMACS/ARCHER2_2022.4_gcc11.md:23]: I think you can delete `module load cpe` as it's loaded by default and only one version is available on ARCHER2 now.

Comment (juanfrh) [reply apps/GROMACS/ARCHER2_2022.4_gcc11.md:23]: You could use `module restore` in case you want a clean module setup.

Comment (RuiApostolo) [reply apps/GROMACS/ARCHER2_2022.4_gcc11.md:23]: I don't think I've meddled with my ~/.basrc, and cpe is not loaded by default (craype is):

```
~❯ ssh archer2
Last login: Fri Jun 16 10:11:24 2023 from 129.215.63.222
#######################################################################################

        @@@@@@@@@
     @@@         @@@            _      ____     ____   _   _   _____   ____    ____
   @@@    @@@@@    @@@         / \    |  _ \   / ___| | | | | | ____| |  _ \  |___ \
  @@@   @@     @@   @@@       / _ \   | |_) | | |     | |_| | |  _|   | |_) |   __) |
  @@   @@  @@@  @@   @@      / ___ \  |  _ <  | |___  |  _  | | |___  |  _ <   / __/
  @@   @@  @@@  @@   @@     /_/   \_\ |_| \_\  \____| |_| |_| |_____| |_| \_\ |_____|
  @@@   @@     @@   @@@
   @@@    @@@@@    @@@       https://www.archer2.ac.uk/support-access/
     @@@         @@@
        @@@@@@@@@

 -         U K R I         -        E P C C        -         H P E   C r a y         -

Hostname:     ln01
Distribution: SLES 15.4 4
CPUS:         256
Memory:       515.3GB
Configured:   2023-06-07

######################################################################################
---------------------------------Welcome to ARCHER2-----------------------------------
######################################################################################

rapostol@ln01:~> module list

Currently Loaded Modules:
  1) craype-x86-rome            4) perftools-base/22.12.0                  7) craype/2.7.19      10) cray-libsci/22.12.1.1  13) epcc-setup-env
  2) libfabric/1.12.1.2.2.0.0   5) xpmem/2.5.2-2.4_3.30__gd0f7936.shasta   8) cray-dsmml/0.2.2   11) PrgEnv-cray/8.3.3      14) load-epcc-module
  3) craype-network-ofi         6) cce/15.0.0                              9) cray-mpich/8.1.23  12) bolt/0.8
rapostol@ln01:~> module load cpe
rapostol@ln01:~> module list

Currently Loaded Modules:
  1) craype-x86-rome            4) xpmem/2.5.2-2.4_3.30__gd0f7936.shasta   7) load-epcc-module   10) cray-dsmml/0.2.2       13) craype/2.7.19
  2) libfabric/1.12.1.2.2.0.0   5) bolt/0.8                                8) PrgEnv-cray/8.3.3  11) cray-libsci/22.12.1.1  14) perftools-base/22.12.0
  3) craype-network-ofi         6) epcc-setup-env                          9) cce/15.0.0         12) cray-mpich/8.1.23      15) cpe/22.12

```

**Comment by RuiApostolo** [comment]
I have explicitly set the cpe version, to future-proof against new cpe versions

**Pull request merged**